openpgp/key_generation.go

```diff
@@ -34,20 +34,28 @@ func NewEntity(name, comment, email string, config *packet.Config) (*Entity, err
 		return nil, err
 	}
 	primary := packet.NewSignerPrivateKey(creationTime, primaryPrivRaw)
+	if config != nil && config.V5Keys {
+		primary.UpgradeToV5()
+	} else {
+		primary.Version = 4
+	}
```

Comment:
Is this needed, since you set the version to 4 in all the low-level functions?

Reply:
Not anymore :+1: